fenix/app/src/main/java/org/mozilla/fenix/browser/StandardSnackbarErrorBinding.kt

```diff
@@ -0,0 +1,97 @@
+
+    override suspend fun onState(flow: Flow<AppState>) {
+        flow.map { state -> state.standardSnackbarError }
+            .distinctUntilChanged()
+            .collect { standardSnackbarError ->
+                if (standardSnackbarError?.standardSnackbarErrorVisibility == true) {
+                    activity.getRootView()?.let { view ->
+                        val snackBar = FenixSnackbar.make(
+                            view = view,
+                            duration = Snackbar.LENGTH_INDEFINITE,
+                            isDisplayedWithBrowserToolbar = true,
+                        )
+                        snackBar.setText(
+                            standardSnackbarError.standardSnackbarErrorMessage
+                                ?: activity.getString(R.string.standard_snackbar_error_message),
+                        )
+                        snackBar.setButtonTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_primary,
+                            ),
+                        )
+                        snackBar.setBackground(
+                            AppCompatResources.getDrawable(
+                                activity,
+                                R.drawable.standard_snackbar_error_background,
+                            ),
+                        )
+                        snackBar.setSnackBarTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_warning,
+                            ),
+                        )
+                        snackBar.setAction(
+                            text = activity.getString(R.string.standard_snackbar_error_dismiss),
+                            action = {
+                                view.context.components.appStore.dispatch(
+                                    AppAction.ShowStandardSnackbarErrorAction(
+                                        StandardSnackbarError(
+                                            false,
+                                            activity.getString(
+                                                R.string.unable_to_save_to_pdf_error,
+                                            ),
+                                        ),
+                                    ),
+                                )
+                            },
+                        )
+                        snackBar.show()
+                    }
+                }
+            }
+    }
+}
+
+/**
+ * Standard Snackbar Error data class.
+ *
+ * @param standardSnackbarErrorMessage that will appear on the snackbar.
+ * @param standardSnackbarErrorVisibility of the snackbar.
+ */
+data class StandardSnackbarError(
+    val standardSnackbarErrorVisibility: Boolean = false,
+    val standardSnackbarErrorMessage: String? = null,
```

Comment:
nit: as these are properties of `StandardSnackbarError` we could just remove the prefix `standardSnackbarError.*`, for example `standardSnackbarErrorVisibility` could be just visibility and `standardSnackbarErrorMessage` could be `message`.

Reply:
Done